### ISSUE TITLE
Flattening imports MUST take care of javascript reserved words

### ISSUE THREAD

**Issue by matthiasg** [rollup/rollup#2680]
- Rollup Version: v1.2.2
- Operating System (or Browser): Windows 10
- Node Version: 10.15.0

### How Do We Reproduce?
See https://github.com/matthiasg/rollup-reserved-word-issue/

Basically importing like this `import * as express from 'express'` and then using `express.static` does not work since the call is flattened to just `static` which is a reserved word.

### Expected Behavior
The resulting file should either leave the `module.static` call or rename `static` to a non-reserved word

### Actual Behavior
```
index.js → stdout...
[!] Error: The keyword 'static' is reserved
module.js (1:16)
1: export function static(text) {
                   ^
2:   console.log(text)
3: }
Error: The keyword 'static' is reserved
    at error (C:\Users\mgt57\AppData\Roaming\npm-cache\_npx\12096\node_modules\rollup\dist\rollup.js:3597:30)
    at Module.error (C:\Users\mgt57\AppData\Roaming\npm-cache\_npx\12096\node_modules\rollup\dist\rollup.js:14376:9)
    at tryParse (C:\Users\mgt57\AppData\Roaming\npm-cache\_npx\12096\node_modules\rollup\dist\rollup.js:14039:16)
    at Module.setSource (C:\Users\mgt57\AppData\Roaming\npm-cache\_npx\12096\node_modules\rollup\dist\rollup.js:14095:35)
    at C:\Users\mgt57\AppData\Roaming\npm-cache\_npx\12096\node_modules\rollup\dist\rollup.js:17409:20
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at findNodeScript.then.existing (C:\Users\mgt57\AppData\Roaming\npm\node_modules\npm\node_modules\libnpx\index.js:268:14)
```

index.js
```javascript
import * as module from './module'

module.static('test')
```

module.js
```javascript
export function static(text) {
  console.log(text)
}
```

**Comment by gund** [comment]
Agreed to this. I hit same bug rendering compiled bundles broken.

Here is [simple example](https://rollupjs.org/repl?version=1.1.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHQlMjBmcm9tJTIwJ2h0dHBzJTNBJTJGJTJGdW5wa2cuY29tJTJGaW8tdHMlNDAxLjguMSUyRmxpYiUyRmluZGV4LmpzJyU1Q24lNUNuY29uc29sZS5sb2codC5udWxsKSUzQiU1Q25jb25zb2xlLmxvZyh0LnVuZGVmaW5lZCklM0IlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=).

I'm using `io-ts` library inside my own library and it has exports named `undefined` and `null`.

While `undefined` is being renamed to `undefined$1`, `null` is kept not renamed so generated code is invalid.

**Comment by matthiasg** [comment]
BTW: Workaround right now is a little involved.

in my case

```
const staticPropertyName = 'static'
express[staticPropertyName](...)
```

It has to be indirected via the variable, calling `express['static']` gets resolved and renaming on import as in `import { static as renamedStatic } from 'express'` also gets resolved and thus doesn't work.

**Comment by lukastaegert** [comment]
Hi, thanks for this issue. As a matter of fact, I am currently working on a complete revamp of the variable resolution/deconflicting logic where it would be easy to incorporate these fixes. Will still take a little while, though.

**Comment by lukastaegert** [comment]
Possible fix a #2689
